Assets/Scripts/Samplers/OpticalRaycaster.cs

```diff
@@ -0,0 +1,120 @@
+    public class OpticalRaycaster : MonoBehaviour
+    { 
+        [SerializeField] private Camera depthCamera;
+        float totalDepth = 0;
+        private Texture2D samplerTexture;
+        
+        [Header("Events")]
+        [SerializeField] public UnityEvent<Vector3> OnDepthSampled;
+
+        void Start()
+        {
+            if(depthCamera.targetTexture == null)
+            {
+                Debug.Log("Depth camera has no target texture. Please assign a render texture to the depth camera.",this.gameObject);
+                this.enabled = false;
+                return;
+            }
+
+            //We will only render on demand using camera.Render()
+            depthCamera.enabled = false; 
+
+            //Create a red channel texture that we can sample depth from
+            samplerTexture = new Texture2D(depthCamera.targetTexture.width, depthCamera.targetTexture.height, TextureFormat.R16, false);
+        }
+
+        private void OnDestroy() {
+            Destroy(samplerTexture);
+        }
+
+        /// <summary>
+        /// Only use this method if it is used continiously in Update.
+        /// If one sample is needed, use AlignDepthCameraToScreenPoint, and GetSamplerCameraWorldPoint
+        /// in a Coroutine with a WaitForEndOfFrame between every step.
+        /// </summary>
+        /// <returns></returns>
+        public Vector3 GetWorldPointAtCameraScreenPoint(Camera camera, Vector3 screenPoint)
+        {
+            AlignDepthCameraToScreenPoint(camera, screenPoint);
+            RenderDepthCamera();
+            
+            return GetDepthCameraWorldPoint();
+        }
+
+        public Vector3 GetWorldPointFromPosition(Vector3 position, Vector3 direction)
+        {
+            AlignDepthCameraFromPositionToDirection(position, direction);
+            RenderDepthCamera();
+
+            return GetDepthCameraWorldPoint();
+        }
+
+        public void AlignDepthCameraToScreenPoint(Camera camera, Vector3 screenPoint)
+        {
+            //Align and rotate sampler camera to look at screenpoint
+            depthCamera.transform.position = camera.transform.position;
+            depthCamera.transform.LookAt(camera.ScreenToWorldPoint(new Vector3(screenPoint.x, screenPoint.y, camera.nearClipPlane)));
+        }
+
+        public void AlignDepthCameraFromPositionToDirection(Vector3 position, Vector3 direction)
+        {
+            //Align depth camera 
+            depthCamera.transform.SetPositionAndRotation(position, Quaternion.LookRotation(direction));
+        }
+
+        public Vector3 GetDepthCameraWorldPoint()
+        {
+            CalculateAverageDepth();
+
+#if UNITY_EDITOR || !UNITY_WEBGL
+            //WebGL/OpenGL2 raw depth value is inverted
+            totalDepth = 1 - totalDepth;
+#endif
+
+            //Move far clip plane according to camera height to maintain precision
+            depthCamera.farClipPlane = depthCamera.transform.position.y * 2.0f;
```

Comment:
misschien iets ophogen of minimum-waarde? edge-case: klein objectje plaatsen van laag boven de grond in de zuidplaspolder(-8m nap) resulteert mogelijk in een negatieve waarde.

Reply:
@mvangog Inderdaad, goeie! -, ik zal m' een minimale waarde geven 👍